imports/math/shared.lua

```diff
@@ -134,4 +134,46 @@ function math.clamp(val, lower, upper) -- credit https://love2d.org/forums/viewt
     return math.max(lower, math.min(upper, val))
 end
 
+---Linearly interpolates between two values over a specified duration.
+---@param start number|table|vector2|vector3|vector4  -- The starting value of the interpolation.
+---@param finish number|table|vector2|vector3|vector4 -- The ending value of the interpolation.
+---@param duration number                             -- The duration over which to interpolate.
+---@return function                                   -- a function that returns interpolated values over time.
+function math.lerp(start, finish, duration)
+	local startTime = GetGameTimer()
+	local typeStart = type(start)
+
+	if typeStart ~= type(finish) then
+		lib.print.error("start and finish must be of the same type")
+	end
+
+	local interpolate = function(s, f, t)
+		if type(s) == "number" or type(s) == "vector2" or type(s) == "vector3" or type(s) == "vector4" then
+			return s + (f - s) * t
+		elseif type(s) == "table" then
+			local result = {}
+			for i in pairs(s) do
+				result[i] = s[i] + (f[i] - s[i]) * t
+			end
+			return result
+		else
+			lib.print.error("unsupported type for lerp: " .. type(s))
+		end
+	end
+
+	local last = false
+	return function()
+		if last then return nil end
+		Wait(0)
+		local t = (GetGameTimer() - startTime) / duration
+		t = math.min(t, 1)
+		if t == 1 then
+			last = true
+			return finish
+		end
+		return interpolate(start, finish, t)
```

Comment:
This is going to type-check on each frame which, while minimal, seems pretty unnecessary.